lib/Sema/CSGen.cpp

```diff
@@ -1716,8 +1716,13 @@ namespace {
         CS.recordFix(
             IgnoreInvalidASTNode::create(CS, CS.getConstraintLocator(locator)));
 
-        return CS.createTypeVariable(CS.getConstraintLocator(repr),
-                                     TVO_CanBindToHole);
+        // Immediately bind the result to a hole since we know it's invalid and
+        // want it to propagate rather than allowing other type variables to
+        // become holes.
+        auto *tv = CS.createTypeVariable(CS.getConstraintLocator(repr),
+                                         TVO_CanBindToHole);
+        CS.recordTypeVariablesAsHoles(tv);
```

Comment:
TypeRepr is an originator for a placeholder type, could we avoid allocating a type variable here?

Reply:
Hmm yeah, let me look into that in a follow-up

Reply:
https://github.com/swiftlang/swift/pull/84030